### DISCUSSION TITLE
Rbatis not creating tables

### DISCUSSION THREAD

**Discussion by OpsRJ** [rbatis/rbatis#71] [General]
I just discovery the project and also I'm very new to Rust. I'm able to connect to the database but the tables are not being created. Do I have to create it manually?

`
pub async fn init_db() -> Rbatis {
    fast_log::init_log("requests.log", 1000,log::Level::Info,None, true);
    let rb = Rbatis::new();
    rb.link("postgres://postgres:######@###.##.#.##:####/study_notes").await.unwrap();

    return rb;
}
`

And inside the function I doing the following:

`
pub async fn save(usr: &User) -> Option<i64> {
    let con = db::init_db().await;

    let saved_sr = con.save("",usr).await.unwrap();

    saved_sr.last_insert_id
}
`

What am I doing wrong?

**Comment by zhuxiujia** [comment]
> I just discovery the project and also I'm very new to Rust. I'm able to connect to the database but the tables are not being created. Do I have to create it manually?
> 
> `
> pub async fn init_db() -> Rbatis {
> fast_log::init_log("requests.log", 1000,log::Level::Info,None, true);
> let rb = Rbatis::new();
> rb.link("postgres://postgres:######@###.##.#.##:####/study_notes").await.unwrap();
> 
> ```
> return rb;
> ```
> 
> }
> `
> 
> And inside the function I doing the following:
> 
> `
> pub async fn save(usr: &User) -> Option {
> let con = db::init_db().await;
> 
> ```
> let saved_sr = con.save("",usr).await.unwrap();
> 
> saved_sr.last_insert_id
> ```
> 
> }
> `
> 
> What am I doing wrong?

*  you can use docker create an  postgres database for command:
```rust
docker run -d --name postgres  -e POSTGRES_PASSWORD=123456 -p 5432:5432 -d postgres
```
*  the [lazy_static](https://crates.io/crates/lazy_static)  crate  is recommended to store RBatis
*  I recommend familiarizing yourself with Rust and then reading the documentation site  [WebSite](https://rbatis.github.io/rbatis.io/#/en/)
*  you can see example/format_test.rs  this example  how to create table
```rust
///this is postgres  database  !
#[cfg(test)]
mod test {
    use rbatis::crud::CRUD;
    use rbatis::rbatis::Rbatis;
    use std::str::FromStr;
    use uuid::Uuid;

    //'formats_pg' use postgres format
    //'id' ->  table column 'id'
    //'{}::uuid' -> format data str
    #[crud_enable(formats_pg: "id:{}::uuid")]
    #[derive(Clone, Debug)]
    pub struct BizUuid {
        pub id: Option<Uuid>,
        pub name: Option<String>,
    }

    /// you may should use pg database! this is docker command for example:
    /// docker run -d --name postgres  -e POSTGRES_PASSWORD=123456 -p 5432:5432 -d postgres
    ///
    #[async_std::test]
    pub async fn test_postgres_uuid() {
        fast_log::init_log("requests.log", 1000, log::Level::Info, None, true);
        let rb = Rbatis::new();
        rb.link("postgres://postgres:123456@localhost:5432/postgres")
            .await
            .unwrap();

        let uuid = Uuid::from_str("df07fea2-b819-4e05-b86d-dfc15a5f52a9").unwrap();
        //create table
        rb.exec(
            "",
            "CREATE TABLE biz_uuid( id uuid, name VARCHAR, PRIMARY KEY(id));",
        )
        .await;
        //insert table
        rb.save(
            "",
            &BizUuid {
                id: Some(uuid),
                name: Some("test".to_string()),
            },
        )
        .await;
        //update table
        rb.update_by_id(
            "",
            &mut BizUuid {
                id: Some(uuid.clone()),
                name: Some("test_updated".to_string()),
            },
        )
        .await;
        //query table
        let data: BizUuid = rb.fetch_by_id("", &uuid).await.unwrap();
        println!("{:?}", data);
        //delete table
        rb.remove_by_id::<BizUuid>("", &uuid).await;
    }
}

```
